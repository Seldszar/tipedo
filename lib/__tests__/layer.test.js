const Layer = require("../layer");

describe("Layer", () => {
  let layer;

  beforeEach(() => {
    layer = new Layer("lorem", {
      lorem: "from lorem",
    });
  });

  test("should return an empty layer", () => {
    expect(new Layer("lorem")).toMatchSnapshot();
  });

  test("should return the `lorem` node", () => {
    expect(layer.get("lorem")).toMatchSnapshot();
  });

  test("should indicate if the `lorem` node exists", () => {
    expect(layer.has("lorem")).toMatchSnapshot();
  });

  test("should return `true` if the node exists", () => {
    expect(layer.delete("lorem")).toMatchSnapshot();
  });

  test("should return `false` if the node doesn't exist", () => {
    expect(layer.delete("ipsum")).toMatchSnapshot();
  });

  test("should clear the layer", () => {
    layer.clear();

    expect(layer).toMatchSnapshot();
  });

  test("should walk through the layer and return the nodes", () => {
    const callback = jest.fn();

    layer.forEach(callback);

    expect(callback).toHaveBeenCalled();
    expect(callback.mock.calls).toMatchSnapshot();
  });

  test("should return the keys", () => {
    expect([...layer.keys()]).toMatchSnapshot();
  });

  test("should return the values", () => {
    expect([...layer.values()]).toMatchSnapshot();
  });

  test("should return the entries", () => {
    expect([...layer.entries()]).toMatchSnapshot();
  });

  test("should return the JSON compatible object", () => {
    expect(layer.toJSON()).toMatchSnapshot();
  });

  test("should return the identity", () => {
    expect(layer.toString()).toMatchSnapshot();
  });

  test("should return an iterator", () => {
    const iterator = layer[Symbol.iterator]();

    expect(iterator).toBeInstanceOf(Object);
    expect(iterator.next).toBeInstanceOf(Function);
  });
});
