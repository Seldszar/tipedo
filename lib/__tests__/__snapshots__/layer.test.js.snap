// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layer should clear the layer 1`] = `
Object {
  "name": "lorem",
  "nodes": Object {},
}
`;

exports[`Layer should indicate if the \`lorem\` node exists 1`] = `true`;

exports[`Layer should return \`false\` if the node doesn't exist 1`] = `false`;

exports[`Layer should return \`true\` if the node exists 1`] = `true`;

exports[`Layer should return an empty layer 1`] = `
Object {
  "name": "lorem",
  "nodes": Object {},
}
`;

exports[`Layer should return the \`lorem\` node 1`] = `"from lorem"`;

exports[`Layer should return the JSON compatible object 1`] = `
Object {
  "name": "lorem",
  "nodes": Object {
    "lorem": "from lorem",
  },
}
`;

exports[`Layer should return the entries 1`] = `
Array [
  Array [
    "lorem",
    "from lorem",
  ],
]
`;

exports[`Layer should return the identity 1`] = `"[object Layer]"`;

exports[`Layer should return the keys 1`] = `
Array [
  "lorem",
]
`;

exports[`Layer should return the values 1`] = `
Array [
  "from lorem",
]
`;

exports[`Layer should walk through the layer and return the nodes 1`] = `
Array [
  Array [
    "from lorem",
    "lorem",
    Map {
      "lorem" => "from lorem",
    },
  ],
]
`;
